planner/solver.go

```diff
@@ -231,18 +224,20 @@ func createPlanningSolutionCandidate(placeIndexes []int, placeClusters [][]match
 	return res, nil
 }
 
-func FindBestPlanningSolutions(ctx context.Context, placeClusters [][]matching.Place, topSolutionsCount int, iterator *MultiDimIterator) ([]PlanningSolution, error) {
+func FindBestPlanningSolutions(ctx context.Context, placeClusters [][]matching.Place, topSolutionsCount int, iterator *MultiDimIterator) (resp chan PlanningResp) {
```

Comment:
we can explicitly specify that `resp` is a **send-only** channel in the function definition, so we prevent read operation in the function.

(resp chan<- PlanningResp)

Reply:
if we do that then the caller `generateSolutions` cannot use the channel.